addons/hr_recruitment/migrations/9.0.1.0/pre-migration.py

```diff
@@ -5,9 +5,31 @@
 
 
 column_renames = {
-    'hr_applicant': [('priority', None)]}
+    'hr_applicant': [
+        ('priority', None),
+        ('availability', None),
+    ]}
+
+
+def migrate_applicant_source(cr):
```

Comment:
You should do this on post-migration, as utm.source doesn't have yet the new structure.